### DISCUSSION TITLE
List Shipping Options for Cart should be based on Stock Locations by default

### DISCUSSION THREAD

**Discussion by joekendal** [medusajs/medusa#10786] [Feature Requests]
Carts by default return all shipping options for all locations. There should be a way to determine which shipping options are available based on the product variant inventory.

The only way I've found around this is to use the `listShippingOptionsForCartWorkflow` which adds extra latency. 

It would make sense if there was 1st class support for this by default as they are core modules from Medusa.

**Comment by olivermrbl** [comment]
@joekendal, what exactly would you like to be able to do, if not use the workflow?

**Reply by joekendal** [reply]
I made a custom override but think it would have made sense as the default Medusa behaviour
